File: go/pangram/pangram.go
package pangram

import "unicode"

const (
	testVersion = 1
	allChars    = 0x03ffffff
)

func IsPangram(s string) bool {
	var set int32

	for _, r := range s {
		if lr := unicode.ToLower(r); lr >= 'a' && lr <= 'z' {
			set |= 1 << (byte(lr) - 'a')
		}
	}

	return allChars == set
}

File: javascript/pangram/pangram.js
let Pangram = function(text) {
  this.text = text

  this.isPangram = () => {
    return (
      new Set(Array.from(this.text.toLowerCase().replace(/[^a-z]/g, "")))
        .size === 26
    )
  }
}

module.exports = Pangram
